app/controllers/api/v2/json_web_tokens_controller.rb

```diff
@@ -0,0 +1,47 @@
+module Api
+  module V2
+    class JsonWebTokensController < V2::BaseController
+      include Api::Version2
+
+      before_action :find_user
+
+      api :GET, "/users/:id/json_web_tokens", N_("Generate JSON web token")
+      param :id, String, required: true
+      param :expires_at, DateTime, desc: N_("Expiration date of the token. If not set, token will never expire")
+      def generate
+        render json: { json_web_token: @user.jwt_token!(expiration: expires_at) }
+      end
+
+      api :PUT, "/users/:id/json_web_tokens", N_("Invalidates all user's JSON web tokens")
```

Comment:
should this be DELETE?

Reply:
In original PR https://github.com/theforeman/foreman/pull/8452 @ares suggested to have it as `PUT` ([comment](https://github.com/theforeman/foreman/pull/8452#discussion_r617451615)), as we do not delete any `json_web_tokens` record but kinda update the user.

This is same as the `GET` action, no strong opinions here.

Reply:
Changing the user record is an implementation detail but for the end user you are effectively removing all tokens. I think DELETE makes sense.